numba/cpython/setobj.py

```diff
@@ -1188,10 +1235,13 @@ def set_constructor(context, builder, sig, args):
 
     # If the argument has a len(), preallocate the set so as to
     # avoid resizes.
+    # both `add` and `for_iter` incref each item in the set, so manually decref
+    # the items to avoid a leak from the double incref
```

Comment:
Unfortunately in this case and in the `set_update` case the item must be decref-ed because each item could occur multiple times. The decref is more a response to `for_iter`'s incref than it is to `add`'s incref